eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -73,22 +73,38 @@ foreach($moniker in $monikers) {
   $onboardedPackages = &$GetOnboardedDocsMsPackagesForMonikerFn `
     -DocRepoLocation $DocRepoLocation -moniker $moniker
   $csvMetadata = @()
+
   foreach($metadataEntry in $fullMetadata) {
     if ($metadataEntry.Package -and $metadataEntry.Hide -ne 'true') {
       $pkgKey = GetPackageKey $metadataEntry
-      if($onboardedPackages.ContainsKey($pkgKey)) {
-        if ($onboardedPackages[$pkgKey] -and $onboardedPackages[$pkgKey].DirectoryPath) {
-          if (!($metadataEntry.PSObject.Members.Name -contains "DirectoryPath")) {
-            Add-Member -InputObject $metadataEntry `
-              -MemberType NoteProperty `
-              -Name DirectoryPath `
-              -Value $onboardedPackages[$pkgKey].DirectoryPath
-          }
-        }
+
+      if (!$onboardedPackages.ContainsKey($pkgKey)) {
+        continue
+      }
+
+      $package = $onboardedPackages[$pkgKey]
+
+      if (!$package) {
```

Comment:
Given the check to make sure the key exists do you ever expect the value to be null?

Reply:
Yes. The value can be null in cases where a package is onboarded in the ci configuration for the docs but that package has no corresponding metadata JSON file. 

We should clean those up, too. https://github.com/Azure/azure-sdk-tools/issues/6283